integration-tests/src/mock_network/setup.rs

```diff
@@ -97,6 +141,131 @@ pub fn setup_mock_network(
     #[cfg(feature = "test_features")]
     let adv = Arc::new(std::sync::RwLock::new(AdversarialControls::default()));
 
+    // if no start height is provided for the mock network at ProduceNewBlocks mode, fall back to
+    // client start height
+    let mode = if let MockNetworkMode::ProduceNewBlocks(None) = mode {
+        MockNetworkMode::ProduceNewBlocks(client_start_height.clone())
+    } else {
+        mode
+    };
+
+    // set up client dir to be ready to process blocks from client_start_height
+    if let Some(start_height) = client_start_height {
+        info!(target:"mock_network", "Preparing client data dir to be able to start at the specified start height {}", start_height);
+        #[cfg(feature = "mock_network")]
+        let mut chain_store =
+            ChainStore::new(client_runtime.get_store(), config.genesis.config.genesis_height);
+        let mut network_chain_store =
+            ChainStore::new(mock_network_runtime.get_store(), config.genesis.config.genesis_height);
+
+        let network_tail_height = network_chain_store.tail().unwrap();
+        let network_head_height = network_chain_store.head().unwrap().height;
+        info!(target:"mock_network",
+              "network data chain tail height {} head height {}",
+              network_tail_height,
+              network_head_height,
+        );
+        assert!(
+            start_height <= network_head_height && start_height >= network_tail_height,
+            "client start height {} is not within the network chain range [{}, {}]",
+            start_height,
+            network_tail_height,
+            network_head_height
+        );
+        let hash = network_chain_store.get_block_hash_by_height(start_height).unwrap();
+        if !mock_network_runtime.is_next_block_epoch_start(&hash).unwrap() {
+            panic!("start height must be the last block of an epoch");
+        }
+
+        // copy chain info
+        #[cfg(feature = "mock_network")]
```

Comment:
shouldn't the whole setup_mock_network be under 'mock_network' feature?

Reply:
yes, but I was having some issues that I don't remember now.

I plan to move the mock_network crate outside of integration-tests. That will make the feature setup cleaner.